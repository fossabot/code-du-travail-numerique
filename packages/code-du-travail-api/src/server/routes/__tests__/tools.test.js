const request = require("supertest");
const Koa = require("koa");
const router = require("../tools");

const app = new Koa();
app.use(router.routes());

it("returns all tools", async () => {
  const response = await request(app.callback()).get(`/api/v1/tools`);
  expect(response.status).toBe(200);
  expect(response.body).toMatchSnapshot();
});

it("returns a single tool", async () => {
  const response = await request(app.callback()).get(
    `/api/v1/tools/this-is-a-second-tool`
  );
  expect(response.status).toBe(200);
  expect(response.body).toMatchSnapshot();
});
