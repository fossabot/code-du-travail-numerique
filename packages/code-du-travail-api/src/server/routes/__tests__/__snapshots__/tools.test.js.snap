// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a single tool 1`] = `
Object {
  "description": "This is a second description",
  "slug": "this-is-a-second-tool",
  "title": "This is a second tool",
}
`;

exports[`returns all tools 1`] = `
Object {
  "children": Array [
    Object {
      "description": "This is a first description",
      "slug": "this-is-a-first-tool",
      "title": "This is a first tool",
    },
    Object {
      "description": "This is a second description",
      "slug": "this-is-a-second-tool",
      "title": "This is a second tool",
    },
  ],
}
`;
